chainer/utils/experimental.py

```diff
@@ -0,0 +1,40 @@
+import chainer
+import inspect
+import warnings
+
+
+def experimental(api_name=None):
+    """A function for marking APIs as experimental
+
+    Developer of the API can mark it as *experimental* by calling
+    this function. When users call experimental APIs, `FutureWarning`
+    is raised once for each experimental API.
+    The presentation of `FutureWarning` is disabled by setting
+    `chainer.disable_experimental_warning` to `True`,
+    which is `False` by default.
+
```

Comment:
As I wrote as unit tests, I expect developers call `utils.experimental` in `__init__` if a class itself is experimental.  We can emit warnings whenever users instantiate the class. Of course, this usage cannot support the case where users call static methods and class methods. But I think developers should call `utils.experimental` in them in that case.


But it is simply because I could not find out an appropriate alternative. If there is a clever way to cover all of instance methods, class methods, and static methods, I would like to choose it.